ui/tract-renderer/src/main/java/org/cru/godtools/tract/widget/PageContentLayout.java

```diff
@@ -736,4 +816,46 @@ public LayoutParams(@NonNull final LayoutParams p) {
             this.childType = p.childType;
         }
     }
+
+    //region bounce thread calls
+
+    /**
+     *  This Handler is what is used to created the delayed post of BOUNCE_ANIMATION_RUNNABLE
+     */
+    private static class BounceHandler extends Handler {
+        private final WeakReference<PageContentLayout> mPageContentLayout;
+
+        BounceHandler(PageContentLayout mPageContentLayout) {
+
+            this.mPageContentLayout = new WeakReference<>(mPageContentLayout);
+        }
+
+        private void verifyAndSendDelayedMessage(int message) {
+            if (!hasMessages(message)) {
+                sendEmptyMessageDelayed(message, DELAY_BETWEEN_BOUNCE_ANIMATION);
+            }
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            super.handleMessage(msg);
+
```

Comment:
```
switch(msg.what) {
    case BOUNCE_MESSAGE:
```

switch execution behavior on what message we receive. That way if we ever add in some other type of message processing we can assign it a unique what value and re-use this Handler.

Reply:
Completely forgot about that good catch.